rasa/nlu/tokenizers/tokenizer.py

```diff
@@ -102,6 +102,8 @@ def __init__(self, config: Dict[Text, Any]) -> None:
         self.token_pattern_regex = None
         if token_pattern:
             self.token_pattern_regex = re.compile(token_pattern)
+        # split intent to prefix and suffix greedily, None means don't split
+        self.prefix_separator_symbol = config.get("prefix_separator_symbol")
```

Comment:
You could just pull this out of the config in the `_split_name` method. I think, it doesn't need to be a attribute of the class

Reply:
Not a strong opinion here, but since `_split_name` is a regular method of the class, it can  access to the `self._config` without needing to change its parameters.

Reply:
Good to know about `self._config`, thanks.

My motivation:
The main idea of having a class attribute `self.prefix_separator_symbol` is to make it explicit for the reader that this class operates (relies on) this state (the constructor usually defines it). By using `_config`, we are going to hide it.

It is debatable, so happy to discuss 😊